src/TypeDefinition.js

```diff
@@ -414,7 +414,23 @@ export type NavigationProp<S> = {
   dispatch: NavigationDispatch,
 };
 
-export type EventListener = {
+export type EventType =
+  | 'willFocus'
+  | 'didFocus'
+  | 'willBlur'
+  | 'didBlur'
+  | 'action';
+
+export type NavigationEventPayload = {
+  type: EventType,
+  action: NavigationAction,
+  state: NavigationState,
+  lastState: NavigationState,
+};
+
+export type NavigationEventCallback = (payload: NavigationEventPayload) => void;
+
+export type NavigationEventListener = {
```

Comment:
Rather than a listener, I think this is usually referred to as the 'subscription'

Other than that, I think this is ready to merge!